packages/core/src/utils.ts

```diff
@@ -0,0 +1,16 @@
+export async function asyncTimeout(ms: number, promise: () => Promise<any>): Promise<any> {
```

Comment:
What's about type inference? Like this:
```suggestion
export async function asyncTimeout<T>(ms: number, promise: () => Promise<T>): Promise<T> {
```
Also, I think it is more logical to use just Promise, not async fn for promise arg

Reply:
Thank you for catching that. I've been so messed up with this timeout situation :/

Reply:
what's about this: